ext/opentelemetry-exporter-cloud-monitoring/src/opentelemetry/exporter/cloud_monitoring/__init__.py

```diff
@@ -0,0 +1,136 @@
+import logging
+from typing import Sequence
+
+import google.auth
+from google.api.label_pb2 import LabelDescriptor
+from google.api.metric_pb2 import MetricDescriptor
+from google.cloud.monitoring_v3 import MetricServiceClient
+from google.cloud.monitoring_v3.proto.metric_pb2 import TimeSeries
+
+from opentelemetry.sdk.metrics.export import (
+    MetricRecord,
+    MetricsExporter,
+    MetricsExportResult,
+)
+from opentelemetry.sdk.metrics.export.aggregate import CounterAggregator
+
+logger = logging.getLogger(__name__)
+MAX_BATCH_WRITE = 200
+
+
+# pylint is unable to resolve members of protobuf objects
+# pylint: disable=no-member
+class CloudMonitoringMetricsExporter(MetricsExporter):
+    """ Implementation of Metrics Exporter to Google Cloud Monitoring"""
+
+    def __init__(self, project_id=None, client=None):
+        self.client = client or MetricServiceClient()
+        if not project_id:
+            _, self.project_id = google.auth.default()
+        else:
+            self.project_id = project_id
+        self.project_name = self.client.project_path(project_id)
+        self._metric_descriptors = {}
+
+    def _add_resource_info(self, series: TimeSeries):
+        """ Add Google resource specific information (e.g. instance id, region)
+
+        :param series: ProtoBuf TimeSeries
+        :return:
+        """
+        # TODO: Leverage this better
+
+    def _batch_write(self, series: TimeSeries):
+        """ Cloud Monitoring allows writing up to 200 time series at once
+
+        :param series: ProtoBuf TimeSeries
+        :return:
+        """
+        write_ind = 0
+        while write_ind < len(series):
+            self.client.create_time_series(
+                self.project_name,
+                series[write_ind : write_ind + MAX_BATCH_WRITE],
+            )
+            write_ind += MAX_BATCH_WRITE
+
+    def _get_metric_descriptor(self, record: MetricRecord) -> MetricDescriptor:
```

Comment:
Looks like this can return actually return None
```suggestion
    def _get_metric_descriptor(self, record: MetricRecord) -> Optional[MetricDescriptor]:
```